### ISSUE TITLE
Make view documentation

### ISSUE THREAD

**Issue by theresaanna** [18F/atf-eregs#193]
Document parent child view relationships.

**Comment by theresaanna** [comment]
in progress: ![views](https://cloud.githubusercontent.com/assets/126935/11983671/0666f642-a96a-11e5-9f18-001c277add94.png)

**Comment by cmc333333** [comment]
Probably a good idea to alter the templates and, to a lesser extent, Python code to match these names

**Comment by theresaanna** [comment]
I think I have everything now. Have I missed anything?
![eregsviewsdocs](https://cloud.githubusercontent.com/assets/126935/12026503/2036f46a-ad6f-11e5-917f-59a01a14c48e.png)

**Comment by theresaanna** [comment]
@cmc333333 I agree, we should make things consistent. I think there's a more generic version of the naming swimming around in my head. I'm going to post a list here.

**Comment by theresaanna** [comment]
Potential more generic/specific/clear view naming:

drawer.js -> nav-drawer.js
toc-view.js -> nav-list-view.js
history-view.js -> timeline-nav-view.js
sidebar-view.js -> right-sidebar-view.js
diff-view.js -> timeline-reg-view.js
breakaway-view.js -> associated-content-view.js? full-screen-view.js? idk


I think SxS doesn't bear consideration as it's a CFPB-specific feature.

**Comment by cmc333333** [comment]
Few ideas:
* I think the (universal) landing page has additional types of content not covered here
* I'd vote we remove sxs-list-view and help-view (and possibly definitions-view) in favor of a more generic type + specific templates
* Similarly, I don't think we need a sxs-view; we should be able to rely on the template to fill that space
* Similarly, and more broadly, I don't _think_ we need specific views for reg vs. diff vs. search-results, do we? Why do they need custom JS?

**Comment by theresaanna** [comment]
> Similarly, and more broadly, I don't think we need specific views for reg vs. diff vs. search-results, do we? Why do they need custom JS?
Each of those views has events and behavior that is specific to the type of content. The place you can probably see this easiest is in the `initialize` and `render` methods. `child-view.js` should take care of most of the common functionality, which each of these views extend. Children implement their own versions of those methods as well. I can't envision a good solution that doesn't require some refactoring, at a conceptual level as well as code. Is it worth it?

> I'd vote we remove sxs-list-view and help-view (and possibly definitions-view) in favor of a more generic type + specific templates
So in a plugin world, I might envision something like...
definition-view.js, sxs-list-view.js and help-view.js might exist as part of their plugins (definitions feature, sxs feature, help feature) and extend our core sidebar-module-view.js. Maybe sxs-list-view.js extends sidebar-list-module-view.js, since the list of links to associated content functionality is probably going to be useful. Does this work for what you had in mind?

> I think the (universal) landing page has additional types of content not covered here
Will look.

> Similarly, I don't think we need a sxs-view; we should be able to rely on the template to fill that space
This is another case where I think the view as we have it now should be part of a plugin. The template may fill that space, but there will still be some events to manage no doubt, so a view of some sort will be needed, I think. sxs-view.js in this case becomes something like breakaway-content-view.js.

**Comment by cmc333333** [comment]
> Each of those views has events and behavior that is specific to the type of content.

I don't follow; why is there different JS behavior for these different views?

> Is it worth it?

I think the frontend code could and should be trimmed down a lot; it should be driven by the backend (particularly the template). Until then, we're requiring new plugin authors understand a rather large Backbone architecture just to display a new template.

> So in a plugin world, I might envision something like...

We can simplify it significantly -- I see no reason that the JS needs to know that this list of links is for ATF Rulings, this list is for SxSes, and this list is for FEC AOs. They're all just links to (e.g.) breakaways; their operations can and should be driven by the templates rather than duplicating logic in the frontend.

> sxs-view.js in this case becomes something like breakaway-content-view.js.

I agree -- I'm saying there's no need to _assume_ SxS-specific events. Rather we should start with the assumption that plugins should require no JS changes. Remember, first pass is about styles and templates.

**Comment by theresaanna** [comment]
I think we're describing the same thing. Ideally, a person implements a content type - let's go with rulings - and one possibility for the content type is to have a sidebar list module that outputs links to rulings in context. There will be default functionality  However, I do think, in the current code, if there are plugins, people will probably want to be able to write _some_ custom UI functionality, which will necessitate JS.

So to me, this conversation hasn't been about the first pass for a long time. What @jbarnicle, @jehlers and I have been working on is to my mind the theme layer, which doesn't include JS at all (at this point). This documentation is for other devs to understand what the Backbone view structure is. It happens to be a tool to look at places that need work, too. Sorry, I should have included more context!

My biggest concern with shifting stuff to the back end is that we may be trading on user experience. That said, I agree that having one less layer to be concerned with as a plugin author is something we should aim for. I think there's a loooong game we should play here to improve overall architecture, but the only way I can see toward getting things implemented is taking it piece by piece. But I think we're just talking from two different contexts, because I know you're on board with that concept. :)

**Comment by cmc333333** [comment]
:+1: sorry for taking us down a not-relevant-right-now tangent!

**Comment by theresaanna** [comment]
@cmc333333 No need to apologize! This is all great thinking. I think we should pick it up perhaps on https://github.com/eregs/eRegulations, so that this is all preserved a bit better.
